include/ccf/endpoints/authentication/cose_auth.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/crypto/pem.h"
+#include "ccf/endpoints/authentication/authentication_types.h"
+#include "ccf/entity_id.h"
+#include "ccf/service/signed_req.h"
+
+namespace ccf
+{
+  struct MemberCOSESign1AuthnIdentity : public AuthnIdentity
+  {
+    /** CCF member ID */
+    MemberId member_id;
+
+    /** Member certificate, used to sign this request, described by keyId */
+    crypto::Pem member_cert;
+
+    /** COSE Protected Headers */
+    /** COSE Payload */
+    /** COSE Signature */
```

Comment:
nit: I suppose this is WIP?

Reply:
Yes, this is the first item on the list. I'm fiddling with the detached envelope still.

Reply:
Done now, except the detach, which is non trivial and will take place separately. It's a pure optimisation, no functional impact.